extension/observer/observer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observer
+
+import (
+	"fmt"
+)
+
+// Protocol defines network protocol for container ports.
+type Protocol string
+
+const (
+	// ProtocolTCP is the TCP protocol.
+	ProtocolTCP Protocol = "TCP"
+	// ProtocolUDP is the UDP protocol.
+	ProtocolUDP Protocol = "UDP"
+)
+
+// Endpoint is a service that can be contacted remotely.
+type Endpoint interface {
+	// ID uniquely identifies this endpoint.
+	ID() string
+	// Target is an address or hostname of the endpoint.
```

Comment:
Generally it's an IP address for pods but there's no reason it couldn't be a DNS name which is why I named it generically.

Reply:
OK, so perhaps the comment can say "IP address or hostname". It was not clear to me that this could not be some other form of an address (other than IP address or host name).